src/server/coprocessor/endpoint.rs

```diff
@@ -333,31 +352,21 @@ impl TiDbEndPoint {
 }
 
 impl TiDbEndPoint {
-    fn handle_request(&self, mut t: RequestTask) {
+    pub fn handle_request(&self, mut t: RequestTask, sel: Result<SelectRequest>) {
         t.stop_record_waiting();
         if let Err(e) = t.check_outdated() {
             on_error(e, t);
             return;
         }
-        let tp = t.req.get_tp();
-        match tp {
-            REQ_TYPE_SELECT | REQ_TYPE_INDEX => {
-                let mut sel = SelectRequest::new();
-                if let Err(e) = sel.merge_from_bytes(t.req.get_data()) {
-                    on_error(box_err!(e), t);
-                    return;
-                }
-                let start_ts = sel.get_start_ts();
-                t.start_ts = Some(start_ts);
-                match self.handle_select(&mut t, sel) {
-                    Ok(r) => respond(r, t),
-                    Err(e) => on_error(e, t),
-                }
-            }
-            _ => on_error(box_err!("unsupported tp {}", tp), t),
+        if sel.is_err() {
+            on_error(sel.unwrap_err(), t);
+            return;
```

Comment:
Since `Result::and` would move `sel` here ,and  `sel`  is needed  later? @andelf